add the ability to directly run wasm from runnables

bnjjj opened the pull request:
Here is a proposal to provide the ability to run a wasm module written in Rust directly from runnables in rust-analyzer and vscode.

bjorn3 commented on editors/code/src/tasks.ts:
```diff
@@ -117,7 +140,27 @@ export async function buildCargoTask(
+    name: string,
+    args: string[]
+): Promise<vscode.Task> {
+    const exec = new vscode.ProcessExecution("wasmer", args, definition);
```
I think this should be configurable without a default runner instead of forcing wasmer.

bnjjj replied:
Yes you're right ! I planned to do this

jonas-schievink commented:
`cargo run --target wasm32-wasi` can already be configured to do this

jonas-schievink commented on crates/ide/src/runnables.rs:
```diff
@@ -306,9 +313,10 @@ pub(crate) fn runnable_fn(sema: &Semantics<RootDatabase>, def: hir::Function) ->
     let kind = if name_string == "main" && def.module(sema.db) == root {
         RunnableKind::Bin
+    } else if name_string == "_start" && def.module(sema.db) == root {
+        RunnableKind::WasmBin
```
Is there a more reliable way to detect this? Just checking for `_start` seems fragile

bjorn3 replied:
In case of wasi you normally don't write `_start` either. You write `main` and then wasi-libc provides `_start` just like libc on any other architecture.

bnjjj commented:
I updated this PR. Here is a summary:
Indeed if we are in a `wasm32-wasi` target, as indicated by Jonas we just use `main` function, the difference is that you need to specify the target in `.cargo/config.toml` and also add a cargo runner to `wasmer` or another wasm runtime. 

BUT for a target `wasm32-unknown-unknown` the main is not automatically binded to `_start`. So I added more constraints to suggest a wasm runnable. Now I detect if we have attribute `no_mangle` but also check if we are using `wasm32-unknown-unknown` target and it's a lib crate (not a bin one).

TBH it's very badly documented in the Rust ecosystem that for a `wasm32-wasi` target we can use `cargo run` if your project is correctly configured. I would like to add more documentation about that but I don't really know where I can put this kind of docs ? In the runtime documentation directly ? Or in Rust-analyzer docs ? I'm open to any sugestions if you have ideas :)
